deprecate.dd

```diff
@@ -55,6 +56,31 @@ $(SPEC_S Deprecated Features,
     )
 
 
+$(H3 $(DEPNAME Comma operator))
+    $(P The comma operator (`,`) allows executing multiple expressions and
+        discards them except for the last which is returned.
+        ---
+        int a = 1;
+        int b = 2;
+        bool ret = a == 2, b == 2; // true
+        ---
+
+        It's also common to use the comma operator in for-loop increment
+        statements to allow multiple statements.
+
+        ---
+        for (; !a.empty && !b.empty; a.popFront, b.popFront)
+        ---
+    )
+$(H4 Corrective Action)
+    $(P Usage of the comma operator outside of for-loops should be forbidden.
+    )
+$(H4 Rationale)
+    $(P The comma operator leads to unintended behavior,
+        (e.g. `synchronized (lockA, lockB)` isn't currently implemented, but still
+        compiles "thanks" to the comma operator).
```

Comment:
I feel like we've had this discussion before and you probably know what I mean.  Spec bug, parser bug, whatever.  The point is that even with comma expressions still allowed in the language we could avoid accepting this bug by changing the parser.